server/commands/list_objects.go

```diff
@@ -0,0 +1,234 @@
+package commands
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/openfga/openfga/pkg/logger"
+	"github.com/openfga/openfga/pkg/tuple"
+	serverErrors "github.com/openfga/openfga/server/errors"
+	"github.com/openfga/openfga/storage"
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	MaximumConcurrentChecks = 100 // todo(jon-whit): make this configurable, but for now limit to 100 concurrent checks
+)
+
+type ListObjectsQuery struct {
+	datastore                    storage.OpenFGADatastore
+	logger                       logger.Logger
+	tracer                       trace.Tracer
+	meter                        metric.Meter
+	listObjectsDeadlineInSeconds int
+	listObjectsMaxResults        uint32
+	resolveNodeLimit             uint32
+}
+
+// NewListObjectsQuery creates a ListObjectsQuery
+func NewListObjectsQuery(datastore storage.OpenFGADatastore, tracer trace.Tracer, logger logger.Logger, meter metric.Meter, listObjectsDeadlineInSeconds int, listObjectsMaxResults uint32, ResolveNodeLimit uint32) *ListObjectsQuery {
```

Comment:
Could we refactor this signature so that it's not so painful on the eyes? When there is more than 3 inputs I think it's helpful to use the Options pattern like so:

```
type ListObjectsQueryOption func(*ListObjectsQuery)

func WithTracer(tracer trace.Tracer) ListObjectsQueryOption {
    return func(q *ListObjectsQuery) {
        q.tracer = tracer
    }
}

func NewListObjectsQuery(opts ...ListObjectsQueryOption) {
    l := &ListObjectsQuery{}
    for _, opt := range opts {
        opt(l)
    }

    return l
}
```

Reply:
We have it elsewhere and it's cryptic. Method signatures this long are not great for readability. Clean Code standards would highly discourage this long of a method signature. The parameter pattern/option pattern or a input/output struct pattern would be a cleaner code pattern.

Alternatively, if we could just drop the constructor. `ListObjecstQuery` is a publicly exported struct anyways, and the constructor isn't buying us anything. You may as well just construct a `&ListObjectsQuery{...}` with the fields it needs.

Reply:
Dropped the constructor